### PR TITLE
Onboarding telemetry

### PR DESCRIPTION
**Pull request by shivani170**
# Description

1. Telemetry event added on onboarding 
2. Telemetry API call in external helm chart

Fixes [AB#1123](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/Devtron%20Team/Issues/?workitem=1123)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Tested out external helm chart app details api

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/onboardingGuide/OnboardingGuide.tsx [hunk 1]
```diff
@@ -123,6 +132,8 @@ function OnboardingGuide({
                         <NavLink
                             to={`${URLS.APP}/${URLS.APP_LIST}`}
                             className="guide_skip dc__no-decor cb-5 fw-6 cursor mb-4"
+                            data-posthog={POSTHOG_EVENT_ONBOARDING.SKIP_AND_EXPLORE_DEVTRON}
+                            onClick={handleSkipOnboarding}
                         >
                             Skip and explore Devtron on your own
```
Comment (vivek-devtron) [inline src/components/onboardingGuide/OnboardingGuide.tsx:138]: we should use SKIP_AND_EXPLORE_DEVTRON variable from utils

File: src/components/onboardingGuide/OnboardingGuide.tsx [hunk 2]
```diff
@@ -54,6 +53,16 @@ function OnboardingGuide({
         handlePostHogEventUpdate(e, POSTHOG_EVENT_ONBOARDING.PREVIEW)
     }
 
+    const handleSkipOnboarding = () => {
+        if (loginCount === 0) {
```
Comment (vivek-devtron) [inline src/components/onboardingGuide/OnboardingGuide.tsx:57]: why this login count check is here? it should anyway be visible on first load

File: src/components/v2/appDetails/AppDetails.component.tsx [hunk 3]
```diff
@@ -23,6 +24,12 @@ const AppDetailsComponent = ({
     const appDetails = IndexStore.getAppDetails();
     const Host = process.env.REACT_APP_ORCHESTRATOR_ROOT;
 
+    useEffect(() => {
+     if( appDetails?.appType?.toString() === AppType.EXTERNAL_HELM_CHART.toString() && params.appId){
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/AppDetails.component.tsx:28]: Why did we add .toString here?

File: src/components/onboardingGuide/OnboardingGuide.tsx (hunk 2 above)
Comment (shivani170) [reply src/components/onboardingGuide/OnboardingGuide.tsx:57]: What if we are going in the `Getting started` page via `Help`section?

File: src/components/v2/appDetails/appDetails.api.ts [hunk 4]
```diff
@@ -12,6 +12,10 @@ export const getInstalledAppDetail = (_appId: number, _envId: number) => {
     return get(`app/detail?app-id=${_appId}&env-id=${_envId}`);
 };
 
+export const getSaveTelemtry = (appId: string) => {
```
Comment (sohel-rp) [inline src/components/v2/appDetails/appDetails.api.ts:15]: please fix typo

File: src/components/v2/appDetails/AppDetails.component.tsx [hunk 5]
```diff
@@ -23,6 +24,12 @@ const AppDetailsComponent = ({
     const appDetails = IndexStore.getAppDetails();
     const Host = process.env.REACT_APP_ORCHESTRATOR_ROOT;
 
+    useEffect(() => {
+     if( appDetails?.appType === AppType.EXTERNAL_HELM_CHART && params.appId){
```
Comment (sohel-rp) [inline src/components/v2/appDetails/AppDetails.component.tsx:28]: Can you please check whether we want to do it for apps with deploymentAppType -> `helm`

File: src/components/onboardingGuide/onboarding.utils.ts [hunk 6]
```diff
@@ -11,7 +11,8 @@ export const POSTHOG_EVENT_ONBOARDING = {
     CONNECT_CHART_REPOSITORY: 'Connect chart repository',
     TOOLTIP_OKAY: 'Tooltip okay',
     TOOLTIP_DONT_SHOW_AGAIN: 'Tooltip Dont show again',
-    HELP: 'Clicked Help'
+    HELP: 'Clicked Help',
+    SKIP_AND_EXPLORE_DEVTRON : 'SkippedOnboarding',
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/onboardingGuide/onboarding.utils.ts:15]: Actually Cant, this key is already merged from BE

File: src/components/v2/appDetails/AppDetails.component.tsx (hunk 5 above)
Comment (shivani170) [reply src/components/v2/appDetails/AppDetails.component.tsx:28]: No, it is required just in case of externalhelmapp

Comment (sohel-rp) [reply src/components/v2/appDetails/AppDetails.component.tsx:28]: Got it

**Review by sohel-rp — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**